org.eclipse.tm4e.core/src/main/java/org/eclipse/tm4e/core/model/ITokenizationSupport.java

```diff
@@ -29,9 +29,10 @@ public interface ITokenizationSupport {
 
 	TokenizationResult tokenize(String line, @Nullable IStateStack state);
 
-	// add offsetDelta to each of the returned indices
-	// stop tokenizing at absolute value stopAtOffset (i.e. stream.pos() +
-	// offsetDelta > stopAtOffset)
-	TokenizationResult tokenize(String line, @Nullable IStateStack state, Integer offsetDelta, Integer stopAtOffset);
-
+	/**
+	 * @param offsetDelta adds offsetDelta to each of the returned indices
+	 * @param timeLimit duration in milliseconds after which tokenization is stopped
```

Comment:
What about using a Duration type directly?

Reply:
Great idea! Just changed it.